### DISCUSSION TITLE
Manually encrypting a video stored in another cloud stroage and uploading it.

### DISCUSSION THREAD

**Discussion by WINOFFRG** [Azure-Samples/media-services-v3-node-tutorials#28] [Q&A] [answered]
Hi! I want to make use of Azure's Content Protection Service and want to encrypt a video stored locally on my device using Bento4 tools. https://www.bento4.com/developers/dash/encryption_and_drm/#marlin-drm
Once my video is widevine protected, I want to upload it to another cloud strorage. Is that possible? Since in widevine we have KEY & KEYID, Is it possible to get this pair from Azure Portal?

**Comment by johndeu** [comment, marked as answer]
Media Services can be configured to encrypt media content when the media is streamed from a Streaming Endpoint. Content is dynamically encrypted with Common Encryption, with support for license/key delivery using PlayReady, Widevine, and FairPlay DRM systems. Media Services does not support streaming pre-encrypted content from storage and it is not normally possible to use media services as a DRM server without also using Media Services to stream the media content (it can be made to work, but the API was not designed for this use-case).

**Reply by WINOFFRG** [reply]
Thanks for your response, I am actually looking to make it work somehow since for some reason I won't be able to stream my content from Azure Cloud Storage but some other 3rd party source. https://learn.microsoft.com/en-us/azure/media-services/previous/media-services-rest-create-contentkey As per the linked article, I am looking for the exact way to get KEY and KEY ID for myself using which I can manually encrypt and upload my content to third party source.

**Reply by johndeu** [reply]
AMS is not really meant to be a DRM only solution.  I would recommend looking into other services like https://www.ezdrm.com/ that are DRM only specific solutions.
